lib/livebook_web/live/hooks/hub_hook.ex

```diff
@@ -0,0 +1,22 @@
+defmodule LivebookWeb.HubHook do
+  import Phoenix.LiveView
+
+  def on_mount(:default, _params, _session, socket) do
+    if connected?(socket) do
+      Livebook.Hubs.subscribe()
+    end
+
+    socket =
+      socket
+      |> assign(saved_hubs: Livebook.Hubs.fetch_hubs())
+      |> attach_hook(:hubs_subscription, :handle_info, &handle_info/2)
+
+    {:cont, socket}
+  end
```

Comment:
Now we have two ways of doing this: `SidebarHelpers.sidebar_handlers/1` and `LivebookWeb.HubHook`. My suggestion is to convert `sidebar_handlers` into an `on_mount` callback and merge the logic there with the logic in this module.

Then instead of doing `socket |> SidebarHelpers.sidebar_handlers()` inside `def mount`, we can call `on_mount`. But please call `on_mount` in the LiveView themselves, not in the router, because this is mostly a presentation concern (it is not policy/access control concern).